src/main.rs

```diff
@@ -64,25 +68,60 @@ fn main() {
 fn run_init(args: Args) -> Result<(), Error> {
 	let root = args.arg_folder.unwrap();
 	let root_path = Path::new(&root);
-	proton_cli::initialize_project(&root_path)
+
+	let admin_pub_key_path = args.arg_public_key.unwrap();
+	let admin_pub_key = try!(utils::file_as_string(&admin_pub_key_path));
+	
+	proton_cli::initialize_project(&root_path, &admin_pub_key)
 }
 
 fn run_new_user(args: Args) -> Result<(), Error> {
+	let admin_key = args.arg_admin_key.unwrap();
+	let admin_key_path = Path::new(&admin_key);
 	let public_key = args.arg_public_key.unwrap();
 	let public_key_path = Path::new(&public_key);
 	let name = args.arg_name.unwrap();
-	proton_cli::new_user(&public_key_path, &name)
+	proton_cli::new_user(&admin_key_path, &public_key_path, &name)
 }
 
 fn run_id_user(args: Args) -> Result<(), Error> {
 	let private_key = args.arg_private_key.unwrap();
 	try!(proton_cli::id_user(&private_key)
-		.map(|_| Ok(())))
+		.map(|user| {
+			println!("{:?}", user);
+			Ok(())
+		})
+	)
 }
 
 fn run_new_sequence(args: Args) -> Result<(), Error> {
+	let admin_key = args.arg_admin_key.unwrap();
+	let admin_key_path = Path::new(&admin_key);
 	let name = args.arg_name.unwrap();
 	let music_file = args.arg_music_file.unwrap();
 	let music_file_path = Path::new(&music_file);
-	proton_cli::new_sequence(&name, &music_file_path)
+	proton_cli::new_sequence(&admin_key_path, &name, &music_file_path)
+}
+
+#[allow(unused_variables)]
+fn run_list_permissions(args: Args) -> Result<(), Error> {
+	let perm_as_str = proton_cli::permissions_as_string();
+	let permissions = perm_as_str.replace(",", "\n");
```

Comment:
This is kind of weird? I'm going to come back to it later, but why is this returning a whole string and using substitution instead of returning a list of permissions and joining them?